homeassistant/components/xiaomi_miio/number.py

```diff
@@ -0,0 +1,165 @@
+"""Support for Xiaomi Mi Air Purifier and Xiaomi Mi Air Humidifier."""
+from dataclasses import dataclass
+from enum import Enum
+import logging
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.core import callback
+
+from .const import (
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    CONF_MODEL,
+    DOMAIN,
+    FEATURE_FLAGS_AIRHUMIDIFIER,
+    FEATURE_FLAGS_AIRHUMIDIFIER_CA4,
+    FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB,
+    FEATURE_SET_MOTOR_SPEED,
+    KEY_COORDINATOR,
+    KEY_DEVICE,
+    MODEL_AIRHUMIDIFIER_CA1,
+    MODEL_AIRHUMIDIFIER_CA4,
+    MODEL_AIRHUMIDIFIER_CB1,
+    MODELS_HUMIDIFIER,
+)
+from .device import XiaomiCoordinatedMiioEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_MOTOR_SPEED = "motor_speed"
+
+
+@dataclass
+class NumberType:
+    """Class that holds device specific info for a xiaomi aqara or humidifier number controller types."""
+
+    name: str = None
+    short_name: str = None
+    unit_of_measurement: str = None
+    icon: str = None
+    device_class: str = None
+    min: float = None
+    max: float = None
+    step: float = None
+    available_with_device_off: bool = True
+
+
+NUMBER_TYPES = {
+    FEATURE_SET_MOTOR_SPEED: NumberType(
+        name="Motor Speed",
+        icon="mdi:fast-forward-outline",
+        short_name=ATTR_MOTOR_SPEED,
+        unit_of_measurement="rpm",
+        min=200,
+        max=2000,
+        step=10,
+        available_with_device_off=False,
+    ),
+}
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Selectors from a config entry."""
+    entities = []
+    if not config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        return
+    host = config_entry.data[CONF_HOST]
+    token = config_entry.data[CONF_TOKEN]
+    model = config_entry.data[CONF_MODEL]
+    device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+
+    _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
+    if model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
+        device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
+    elif model in [MODEL_AIRHUMIDIFIER_CA4]:
+        device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA4
+    elif model in MODELS_HUMIDIFIER:
+        device_features = FEATURE_FLAGS_AIRHUMIDIFIER
+    else:
+        return
+
+    for feature in NUMBER_TYPES:
+        number = NUMBER_TYPES[feature]
+        if feature & device_features and feature in NUMBER_TYPES:
```

Comment:
I think we can remove this check here too, as in the select platform.

Reply:
Removed feature checks en cleaned up imports